lib/ukbitops/Makefile.uk

```diff
@@ -0,0 +1,20 @@
+$(eval $(call addlib_s,libukbitops,$(CONFIG_LIBUKBITOPS)))
+
+CINCLUDES-$(CONFIG_LIBUKBITOPS)	 		+= -I$(LIBUKBITOPS_BASE)/include
+ASINCLUDES-$(CONFIG_LIBUKBITOPS)	  	+= -I$(LIBUKBITOPS_BASE)/include
+CXXINCLUDES-$(CONFIG_LIBUKBITOPS)		+= -I$(LIBUKBITOPS_BASE)/include
```

Comment:
Since you are defining `CXXINCLUDES` here, you should be adding C++ guards in the headers.